docs/cloudlab.md

```diff
@@ -0,0 +1,77 @@
+# CloudLab getting started guide
+
+## Create CloudLab experiments
+
+First, submit an account request to CloudLab (https://www.cloudlab.us/). Things to note:
+
+- Select "Join Existing Project" and type: `Sieve-Acto`
+- The username and key you provide will be used for SSH login
+
+Wait for the admin to approve your account. Once you are able to login, familiarize yourself with the web-based dashboard, and [the concept of *profiles* and *experiments*](https://docs.cloudlab.us/basic-concepts.html).
+
+Although you should be able to login onto any machine instantiated by your project collaborators (i.e. a Linux user will be automatically created for you on every machine with `authorized_keys` set up), our current practice is to let everyone run code **on their own experiments**.
+
+Launch an experiment via the web dashboard:
+
+1. Click "Experiments" -- "Start Experiment". The default selected profile should be `small-lan`. "Next".
+2. Enter/Choose parameters:
+    - "Select OS image": `UBUNTU 20.04`
+    - "Optional physical node type": `c6420`
+    - Leave other parameters as default. (Especially those regarding temporary filesystem -- we'll deal with it after provisioning using Ansible.)
+3. "Next". Give your experiment a name. "Next". "Next".
+
+Wait for the provisioning to finish. The web dashboard will show you the server address, in the form of `clnodeXXX.clemson.cloudlab.us`.
+
+## Configure CloudLab machines
+
+We are going to manage CloudLab machines with Ansible **from our local machine**. (Ideally this "local machine" can be a CloudLab one too. However, the scripts were not designed to do so and may mess things up. For the time being, minor tweaks are needed in order to let them function properly in such cases.)
+
+**On your local machine**:
+
+```shell
+#
+# Install Ansible
+#
+
+sudo apt update
+sudo apt install software-properties-common
+sudo add-apt-repository --yes --update ppa:ansible/ansible
+sudo apt install ansible
+ansible-galaxy collection install ansible.posix
+ansible-galaxy collection install community.general
+
+#
+# Checkout the repository
+#
+
+cd ~
+git clone https://github.com/xlab-uiuc/acto.git
+cd acto/
+git checkout sosp-ae
+
+#
+# Prepare the CloudLab machine(s) with Ansible
+#
+
+cd scripts/ansible/
+# TODO: replace the placeholders with your real domain and user name you
+# obtained or submitted in the previous section
+domain="clnodeXXX.clemson.cloudlab.us"
+user="alice"
+echo "$domain ansible_connection=ssh ansible_user=$user ansible_port=22" > ansible_hosts
+sed -i 's|git@github.com:xlab-uiuc/acto.git|https://github.com/xlab-uiuc/acto.git|' acto.yaml # FIXME: see #247
```

Comment:
I think let's just fix the acto.yaml under the ansible script for this one

Reply:
Sure I notice the fix has already been there in `soap-ae`. let me port it to the other PR targeting `main`

Reply:
Thank you!

Reply:
Do we also move everything to `~/workdir/` for `main`? @tylergu

Reply:
Yes, because the workdir will be the root of the mounted filesystem, and we cannot clone a repo into a non-empty directory unless we do some hack